cmd/env.go

```diff
@@ -95,18 +100,45 @@ var envAddCmd = &cobra.Command{
 	Short: "Add a new environment to a ksonnet project",
 	RunE: func(cmd *cobra.Command, args []string) error {
 		flags := cmd.Flags()
-		if len(args) != 2 {
-			return fmt.Errorf("'env add' takes two arguments, the name and the uri of the environment, respectively")
+		if len(args) < 1 {
+			return fmt.Errorf("'env add' takes at least one argument, which is the name of the environment")
+		}
+		if len(args) > 2 {
+			return fmt.Errorf("'env add' takes at most two arguments, which is the name and the uri of the environment, respectively")
```

Comment:
I am thinking that we might want the second argument to be context name instead of URI. Then we'd allow people to tweak the environment with `--uri`, `--namespace`, `--context`, and so on. So the form would be something like:

```
$ ks init <app-name> [ctx-name] [--kubefile=<kf>] [--namespace=<ns>] [--uri=<uri>].
```

You wouldn't need a context if you have a `--uri`.

This complicates the logic somewhat, but I think people generally do not want to pass in the URI directly. I don't think most of our users will even know where it is.

Reply:
It's a redesign from the original intention of the issue, but I agree with you -- I think it'll be clearer for users to understand how to initiate environments using the context.

Changing the command line logic is relatively straightforward, but how far do you think we should deviate from the original design -- i.e., should the envrionment's `spec.json` be storing the cluster URI or the context?

Reply:
We talked offline. Decided to change the command line interface for `ks env add` and `ks init` to take optional `--context` and `--uri` flags, but the logic around storing env metadata, i.e. storing `uri` in `spec.json`, will remain unchanged.